shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -445,6 +473,58 @@ - (UITextRange*)lineEnclosingPosition:(UITextPosition*)position {
 
 @end
 
+#pragma mark - FlutterTextSelectionRect
+
+@implementation FlutterTextSelectionRect
+
+@synthesize rect = _rect;
+@synthesize writingDirection = _writingDirection;
+@synthesize containsStart = _containsStart;
+@synthesize containsEnd = _containsEnd;
+@synthesize isVertical = _isVertical;
```

Comment:
I suspect these synthesize statements are superfluous.  You should only have to do this these days if you are implementing a protocol with properties.

Reply:
So, I believe the issue is that `FlutterTextSelectionRect` subclasses `UITextSelectionRect`, which has all of these properties as readonly. From a little bit of reading up on the errors that I get when removing these, I believe I either need this or to mark them as dynamic with e.g. `@dynamic rect;` and so on. It looks like `@dynamic` is used in the `FlutterViewController`, so I will swap to that, assuming that will be the preferred method.

Reply:
As it turns out, swapping this to dynamic breaks the tests - the setter selectors are missing in that case, apparently, so I think we do need the synthesize. This is a similar style to what's used in this file for the tokenizer and textInputDelegate.